beacon-chain/core/peerdas/helpers.go

```diff
@@ -205,39 +206,29 @@ func DataColumnSidecarsForReconstruct(
 	blobKzgCommitments [][]byte,
 	signedBlockHeader *ethpb.SignedBeaconBlockHeader,
 	kzgCommitmentsInclusionProof [][]byte,
-	blobs []kzg.Blob,
+	cellsAndProofs []kzg.CellsAndProofs,
 ) ([]*ethpb.DataColumnSidecar, error) {
-	blobsCount := len(blobs)
+	// Each CellsAndProofs corresponds to a Blob
+	// So we can get the BlobCount by checking the length of CellsAndProofs
+	blobsCount := len(cellsAndProofs)
 	if blobsCount == 0 {
 		return nil, nil
 	}
 
-	// Compute cells and proofs.
-	cells := make([][kzg.CellsPerExtBlob]kzg.Cell, 0, blobsCount)
-	proofs := make([][kzg.CellsPerExtBlob]kzg.Proof, 0, blobsCount)
-
-	for i := range blobs {
-		blob := &blobs[i]
-		blobCells, blobProofs, err := kzg.ComputeCellsAndKZGProofs(blob)
-		if err != nil {
-			return nil, errors.Wrap(err, "compute cells and KZG proofs")
-		}
-
-		cells = append(cells, blobCells)
-		proofs = append(proofs, blobProofs)
-	}
-
```

Comment:
This code has now been moved to recoverCellsAndProofs, so instead of computing the cells and proofs here, we now pass them to this method